docs/Finance/LevyDeclarations/Index.md

```diff
@@ -0,0 +1,133 @@
+
+### Purpose
+This page describes:
+
+-  How levy declarations are loaded into the system
+-  How the HMRC system is used in tests
+-  How test data is created 
+
+> At the time of writing the solution is switching from cloud services to app services, so worker roles will become web jobs for example.  
+     
+# How levy declarations are loaded into the system
+
+### An `EmployerRefreshLevyQueueMessage` is posted to service bus for every account and PAYE Ref
+
+- Once a month (at about 16:00 on the 20th day) a web job called `SFA.DAS.EAS.LevyAccountUpdater.WebJob` is kicked off. 
+- This job runs `AccountUpdater.RunUpdate` which gets all Accounts into memory and for each account fetches all PAYE schemes.
+- An `EmployerRefreshLevyQueueMessage` message is posted to service bus for each account id and PAYE scheme ref. 
+
+### A constantly running worker role subscribes to a topic containing `EmployerRefreshLevyQueueMessage`
+
+- The worker role `SFA.DAS.EAS.LevyDeclarationProvider.Worker` uses our `MessageSubscriberFactory` pattern to subscribe to a topic named ` MA_LevyDeclaration`.
+- Each `EmployerRefreshLevyQueueMessage` message recieved is used to update the database
+ 
+### How each message is processed
+ 
+- English Fractions for the PAYE ref are obtained from the HMRC and merged into the database.
+- Levy Declarations for the PAYE ref are obtained from the HMRC. We only ask for Levy Declarations that are submitted since the last submission we have already. This date is obtained from stored proc `[employer_financial].[GetLastLevyDeclarations_ByEmpRef]`). If this is the first levy declaration then only levy declarations since the 01-Apr-2017 are fetched (this date is hard-wired into the HmrcService).The supplied data is adapted in a couple of ways:
+	- Duplicate submissions are removed from the data supplied by HMRC. This is done based on submisson ID and if duplicates are found one of the duplicates will be arbitrarily selected. _I do not know what the justification for this is_. 
+	- `LevyDueYtd` is nulled out for all declarations where `NoPaymentForPeriod` is true
+	- Any Levy Declarations for period 12 with a date more than 21 days after the start of period 12 are considered to be "year end adjustments" - they are not explicitly declared as such by the HMRC. Any such year end adjustments are exposed to a special process.
+	- ... more to come
+
+# How the HMRC system is used in tests
+
+There is not a test system provided by the HMRC. Instead, a fake HMRC has been created by the SFA, which is hosted externally on `https://das-alpha-hmrc-api-mock.herokuapp.com/apprenticeship-levy`. This system is written in Scala, the source code to which is available here `https://github.com/UKGovernmentBEIS/das-alpha-hmrc-api-mock`. This is a fairly think layer that fronts a managed Mongo database (hosted at MLabs). There is some functionality but this is limited to enforcing token-based authentication and filtering on the document collections. 
```

Comment:
HMRC do have a sandbox system but its a nightmare for us to update it as they have to do it for us so we created our own version

Reply:
think layer => thin layer? or Thick layer?